### PR TITLE
668 opentsdb

### PR DESCRIPTION
**Pull request by zhangzinan**
new agent for opentsdb.

### CODE REVIEW

File: opentsdb/getmetrics_opentsdb.py [hunk 1]
```diff
@@ -24,144 +29,224 @@ def getParameters():
...
     if response.status_code == 200:
-        metricList = response.json()
-        logger.debug("Get metric list from opentsdb: " + str(metricList))
-    return metricList
+        metric_list = response.json()
+        logger.debug("Get metric list from opentsdb: " + str(metric_list))
+    return metric_list
+
+
+def get_metric_list_from_file(config, filePath):
```
Comment (moharnab123saikia) [inline opentsdb/getmetrics_opentsdb.py]: variable naming

File: opentsdb/getmetrics_opentsdb.py [hunk 2]
```diff
@@ -24,144 +29,224 @@ def getParameters():
...
+            m = re.search(r'(?P<metric>\d\.\d\..+):', line)
+            if m:
+                metric = m.groupdict().get('metric')
+                metric_list.add(metric)
+                logger.debug("Get metric list from file: " + str(metric_list))
+    return list(metric_list)
 
 
-def getMetricData(config, metricList, startTime, endTime):
+def get_metric_data(config, metric_list, grouping_map, startTime, endTime):
```
Comment (moharnab123saikia) [inline opentsdb/getmetrics_opentsdb.py]: variable naming

File: opentsdb/getmetrics_opentsdb.py [hunk 3]
```diff
@@ -170,22 +255,22 @@ def chunks(l, n):
...
-def setloggerConfig(logLevel):
+def set_logger_config(level):
     """Set up logging according to the defined log level"""
     # Get the root logger
-    logger = logging.getLogger(__name__)
+    logger_obj = logging.getLogger(__name__)
     # Have to set the root logger level, it defaults to logging.WARNING
-    logger.setLevel(logLevel)
+    logger_obj.setLevel(level)
     # route INFO and DEBUG logging to stdout from stderr
```
Comment (moharnab123saikia) [inline opentsdb/getmetrics_opentsdb.py]: We can add a log formatter to show timestam of the log and log level. Some other agent should have that code. The user has asked for it before

File: opentsdb/getmetrics_opentsdb.py [hunk 4]
```diff
@@ -24,144 +29,224 @@ def getParameters():
...
+                logger.debug("Get metric list from file: " + str(metric_list))
+    return list(metric_list)
 
 
-def getMetricData(config, metricList, startTime, endTime):
+def get_metric_data(config, metric_list, grouping_map, startTime, endTime):
     """Get metric data from Open TSDB API"""
-    openTsdbMetricList = []
+
+    def full_data(d):
```
Comment (moharnab123saikia) [inline opentsdb/getmetrics_opentsdb.py]: function inside a function is a bit less readable.

File: opentsdb/getmetrics_opentsdb.py [hunk 5]
```diff
@@ -24,144 +29,224 @@ def getParameters():
...
+        value_map = {}
+        metric_name = d.get('metric')
```
Comment (moharnab123saikia) [inline opentsdb/getmetrics_opentsdb.py]: the variable name d is not descriptive.

File: opentsdb/getmetrics_opentsdb.py [hunk 6]
```diff
@@ -24,144 +29,224 @@ def getParameters():
...
     else:
-        parameters['chunkSize'] = int(options.chunkSize)
-    return parameters
+        params['chunkSize'] = int(options.chunkSize)
+    return params
 
 
-def getAgentConfigVars():
-    configVars = {}
+def get_agent_config_vars():
```
Comment (yqian10) [inline opentsdb/getmetrics_opentsdb.py]: I think we do not use .agent.bashrc anymore. Instead, we use the config.ini in the agent code folder

File: opentsdb/getmetrics_opentsdb.py [hunk 7]
```diff
@@ -199,41 +284,65 @@ def filter(self, record):
...
+        time_list = [(data_start_ts, data_end_ts)]
+    else:
+        # get data from history date
+        start_day = agent_config['OPENTSDB_HIS_START_DAY']
+        end_day = agent_config['OPENTSDB_HIS_END_DAY']
+        start_day_obj = datetime.strptime(start_day, "%Y-%m-%d")
+        start_ts = int(time.mktime(start_day_obj.timetuple()))
+        end_day_obj = datetime.strptime(end_day, "%Y-%m-%d")
+        end_ts = int(time.mktime(end_day_obj.timetuple()))
+        timeInterval = (end_ts - start_ts) / 60
```
Comment (yqian10) [inline opentsdb/getmetrics_opentsdb.py]: variable naming

File: opentsdb/getmetrics_opentsdb.py [hunk 8]
```diff
@@ -24,144 +29,224 @@ def getParameters():
...
 
 
-def getAgentConfigVars():
-    configVars = {}
+def get_agent_config_vars():
+    config_vars = {}
     with open(os.path.join(parameters['homepath'], ".agent.bashrc"), 'r') as configFile:
-        fileContent = configFile.readlines()
-        if len(fileContent) < 6:
+        file_content = configFile.readlines()
```
Comment (yqian10) [inline opentsdb/getmetrics_opentsdb.py]: variable naming

**Comment by ptmr3** [comment]
Replaced by II-2225

**Pull request closed without merging**